spec.html

```diff
@@ -28139,6 +28402,112 @@ <h1>
+        </emu-grammar>
+        <emu-alg>
+          1. Return the SV of |StringLiteral|.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-with-clause-to-attributes" type="sdo">
+        <h1>Static Semantics: WithClauseToAttributes ( ): a List of ImportAttribute Records</h1>
+        <dl class="header">
+        </dl>
+
+        <emu-grammar>
+          WithClause : `with` `{` `}`
+        </emu-grammar>
+        <emu-alg>
+          1. Return a new empty List.
+        </emu-alg>
+
+        <emu-grammar>
+          WithClause : `with` `{` WithEntries `,`? `}`
+        </emu-grammar>
+        <emu-alg>
+          1. Let _attributes_ be WithClauseToAttributes of |WithEntries|.
+          1. Sort _attributes_ according to the lexicographic order of their [[Key]] fields, treating the value of each such field as a sequence of UTF-16 code unit values. NOTE: This sorting is observable only in that hosts are prohibited from distinguishing among attributes by the order they occur in.
+          1. Return _attributes_.
+        </emu-alg>
+
+        <emu-grammar>WithEntries : AttributeKey `:` StringLiteral</emu-grammar>
+        <emu-alg>
+          1. Let _key_ be the AttributeName of |AttributeKey|.
+          1. Let _entry_ be the ImportAttribute Record { [[Key]]: _key_, [[Value]]: SV of |StringLiteral| }.
+          1. Return « _entry_ ».
+        </emu-alg>
+
+        <emu-grammar>WithEntries : AttributeKey `:` StringLiteral `,` WithEntries</emu-grammar>
+        <emu-alg>
+          1. Let _key_ be the AttributeName of |AttributeKey|.
+          1. Let _entry_ be the ImportAttribute Record { [[Key]]: _key_, [[Value]]: SV of |StringLiteral| }.
+          1. Let _rest_ be WithClauseToAttributes of |WithEntries|.
+          1. Return the list-concatenation of « _entry_ » and _rest_.
+        </emu-alg>
+
+        <emu-note>
+          <p>This section is extended by <emu-xref href="#sec-deprecated-assert-keyword-for-import-attributes"></emu-xref>.</p>
+        </emu-note>
+      </emu-clause>
+
+      <emu-clause id="sec-deprecated-assert-keyword-for-import-attributes" deprecated normative-optional>
+        <h1>Deprecated `assert` keyword for Import Attributes</h1>
+
+        <p>Import Attributes are denoted using the `with` keyword. However, there are existing implementations based on a previous version of the proposal using the `assert` keyword. Due to potential web compatibility risks, the `assert` keyword is still included in this specification, but its removal is being investigated.</p>
+
+        <p>This deprecated feature includes the `assert` fallback for import attributes in import calls (step <emu-xref href="#step-evaluate-import-call-assert-fallback"></emu-xref> of <emu-xref href="#sec-evaluate-import-call"></emu-xref>).</p>
+
+        <p>The following augments the |ImportDeclaration| production in <emu-xref href="#sec-imports"></emu-xref>:</p>
+        <emu-grammar type="definition">
```

Comment:
`AssertClause` is defined here. `ImportDeclaration`'s definition is split across two different `<emu-clause>`s: the LHSs listed here are not defined anywhere else. This is similar to how clause B.1.1 uses `type="definition"`.

Reply:
I think it is different from the Appendix B cases. After generating the `index.html` file using `npm run build`, all the links of `ImportDeclaration` points out to the following definition:
```
        <emu-grammar type="definition">
          ImportDeclaration :
            `import` ImportClause FromClause [no LineTerminator here] AssertClause `;`
            `import` ModuleSpecifier [no LineTerminator here] AssertClause `;`
```
Instead of the following definition:
```
      <emu-grammar type="definition">
        ImportDeclaration :
          `import` ImportClause FromClause `;`
          `import` ModuleSpecifier `;`
          `import` ImportClause FromClause [no LineTerminator here] WithClause `;`
          `import` ModuleSpecifier [no LineTerminator here] WithClause `;`
```
In addition, please look at Appendix A in the generated `index.html`. Unfortunately, it also only listed the following cases:
```
            `import` ImportClause FromClause [no LineTerminator here] AssertClause `;`
            `import` ModuleSpecifier [no LineTerminator here] AssertClause `;`
```
I think it happens because it is the ONLY case that two different `type="definition"` tags for the same nonterminal.
Is it really what you expected? And, are there other examples with two different `type="definition"` tags not in the appendix?

Reply:
I tried to split this in two `emu-grammar`s:
```html
        <emu-grammar>
          ImportDeclaration :
            `import` ImportClause FromClause [no LineTerminator here] AssertClause `;`
            `import` ModuleSpecifier [no LineTerminator here] AssertClause `;`
        </emu-grammar>
        <emu-grammar type="definition">
          AssertClause :
            `assert` `{` `}`
            `assert` `{` WithEntries `,`? `}`
        </emu-grammar>
```

however, it gets rendered with different indentation levels:
<img width="646" alt="image" src="https://github.com/tc39/ecma262/assets/7000710/cb2c179a-2828-461c-be09-78b09248190a">

I agree that the `ImportDeclaration` should not be marked as definition, because we'd want the anchor to link to the "real" definition of ImportDeclaration.

Reply:
Alternatively, mark it as @nicolo-ribaudo originally did (one `emu-grammar` with `type="definition"`) and modify ecmarkup to handle such cases differently (e.g., link occurrences of `ImportDeclaration` to the non-deprecated definition).

There's also the possibility of creating a new `type` value, meaning "this emu-grammar defines additional right-hand-sides for a nonterminal whose 'primary' definition is elsewhere", though I'm doubtful it's necessary.

Reply:
I'm inclined to keep it as two `emu-grammar` elements and fix the CSS by manually tweaking it inline, along the lines of

```html
The following augments the ImportDeclaration production in X.X.X:
<emu-grammar>
  ImportDeclaration :
    `import` ImportClause FromClause [no LineTerminator here] AssertClause `;`
    `import` ModuleSpecifier [no LineTerminator here] AssertClause `;`
</emu-grammar>

The following additional nonterminal is defined:
<emu-grammar type="definition" style="margin-left: 0px">
  AssertClause :
    `assert` `{` `}`
    `assert` `{` WithEntries `,`? `}`
</emu-grammar>
```

Note the explicit `style` attribute.